### ISSUE TITLE
Polling model?

### ISSUE THREAD

**Issue by jackfirth** [vfarcic/docker-flow-swarm-listener#32]
Is there a project that's _like_ this, but runs as a server that offers a single "current list of services" HTTP resource? Possibly with headers / links that let a client of the resource subscribe to future updates? The reason I'd like this is so the swarm listener doesn't have to be configured ahead of time with knowledge of what servers to contact, and so the swarm listener service can scale to multiple containers easily. Also, dependent services wouldn't need to bind mount the Docker socket or speak Docker's protocol to get the current list of services.

**Comment by vfarcic** [comment]
If you publish DFSL port `8080`, you can get the list of all services using the endpoint `/v1/docker-flow-swarm-listener/get-services`.

> The reason I'd like this is so the swarm listener doesn't have to be configured ahead of time with knowledge of what servers to contact, and so the swarm listener service can scale to multiple containers easily. Also, dependent services wouldn't need to bind mount the Docker socket or speak Docker's protocol to get the current list of services.

I'm not sure I understood this fully. Can you provide an example?

**Comment by jackfirth** [comment]
I've confused myself entirely 😄

At the time of opening this issue, I thought:

1. Docker provided a gRPC API, instead of an HTTP API. Not sure how I got this notion.
2. Docker's API could _only_ be exposed over a local socket. But I've discovered [there is an option](https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option) to access it over HTTPS.

So I was hoping for something like DFSL that I could use as a bridge between my services and Docker. But that's entirely unnecessary and I can just use the Docker engine API directly.
